core/src/main/java/org/elasticsearch/index/mapper/core/BooleanFieldMapper.java

```
@@ -198,6 +201,14 @@ public boolean useTermQueryWithQueryString() {
             failIfNoDocValues();
             return new DocValuesIndexFieldData.Builder().numericType(NumericType.BOOLEAN);
         }
+
+        @Override
+        public ValueFormat valueFormat(@Nullable String format, DateTimeZone timeZone) {
+            if (format != null) {
+                throw new IllegalArgumentException("Field [" + name() + "] of type [" + typeName() + "] does not support custom formats");
+            }
+            return new ValueFormat.Boolean();
```

should we also throw an exception if timeZone is specified here?